### PR TITLE
Fix duplicate estimates

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

We noticed that between re-training fresh models we were receiving the same estimates. This was happening because the model updates were failing and we could not detect this error because of https://github.com/valory-xyz/open-aea/issues/424.

The updates were failing because of incorrectly querying the ids from the data, assuming that they contained a `.csv` extension, which was not the case.

The test was implemented incorrectly because it was not passing the pipelines' input with the correct key names, i.e., the `.joblib` extension was not present. As a result, the method was not tested correctly as the `.csv` extension assumption was not being tested.

The test was fixed on de25faa4f86bb727fa9e9da4db3999393ab91678, which made the update method to fail. The next commits fix the issue.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

**Review by 0xArdi — COMMENTED** [review]
LGTM - waiting for the CI.

File: packages/valory/skills/apy_estimation_abci/tests/test_ml/test_forecasting.py [hunk 1]
```diff
@@ -342,25 +342,17 @@ def test_update_forecaster_per_pool(
 
         mismatch = "test" if id_mismatch else ""
         pools_to_dummy_pipelines = {
-            pool_id + mismatch: deepcopy(dummy_pipeline)
+            "".join((pool_id, ".joblib", mismatch)): deepcopy(dummy_pipeline)
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_ml/test_forecasting.py:345]: The test change that revealed the issue.

File: packages/valory/skills/apy_estimation_abci/tests/test_ml/test_forecasting.py [hunk 2]
```diff
@@ -342,25 +342,17 @@ def test_update_forecaster_per_pool(
...
+        if mismatch:
+            assert not any(updated)
         else:
-            update_forecaster_per_pool(
-                prepare_batch_task_result, pools_to_dummy_pipelines
-            )
-            assert all(
-                pipeline.updated for pipeline in pools_to_dummy_pipelines.values()
-            )
+            assert all(updated)
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_ml/test_forecasting.py:355]: Removed code dup.

File: packages/valory/skills/apy_estimation_abci/ml/forecasting.py [hunk 3]
```diff
@@ -305,7 +305,7 @@ def update_forecaster_per_pool(
     :param forecasters: the forecasters to update.
     """
     for id_ in forecasters.keys():
-        apy = y.loc[y["id"] == id_.replace(".joblib", ".csv"), "APY"]
+        apy = y.loc[y["id"] == id_.replace(".joblib", ""), "APY"]
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/ml/forecasting.py:308]: The fix.

File: packages/valory/skills/apy_estimation_abci/behaviours.py [hunk 4]
```diff
@@ -1298,7 +1298,7 @@ def __init__(self, **kwargs: Any) -> None:
         """Initialize Behaviour."""
         super().__init__(**kwargs)
         self._async_result: Optional[AsyncResult] = None
-        self._y: Optional[PoolIdToTrainDataType] = None
+        self._y: Optional[pd.DataFrame] = None
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/behaviours.py:1301]: Type hint was incorrect.

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**